tests/Toolkit/ATest.php

```diff
@@ -265,6 +265,34 @@ public function testLast()
         $this->assertSame('tweet', A::last($this->_array()));
     }
 
+    /**
+     * @covers ::random
+     */
+    public function testRandom()
+    {
+        $array = $this->_array();
+        $arrayKeys = array_flip(array_keys($array));
+        $arrayValues = array_flip(array_values($array));
+
+        // Assert existence and correctness of keys
+        $random1 = A::random($array, 1);
+        $this->assertTrue(in_array(array_values($random1)[0], $array));
+        $this->assertTrue(in_array(array_key_first($random1), array_keys($array)));
+
+        // Assert order of keys in non-shuffled random
+        $random2 = A::random($array, 2);
+        $this->assertTrue($arrayKeys[array_key_first($random2)] < $arrayKeys[array_key_last($random2)]);
+
+        // Assert count in completely shuffled array
+        $random3 = A::random($array, 3, true);
+        $this->assertSame(count($array), count($random3));
```

Comment:
Shouldn't this always be `3`?

```suggestion
        $this->assertCount(3, $random3);
```

Reply:
True, but I wanted highlight the fact that the count doesn't change, so when you look at the test a year later, you don't expect t he count to be three, but rather the same as original value.

Reply:
But if the input contains 4 items, won't the test fail then? I mean after all the expected behavior is to get the number of results that were requested, right?